### PR TITLE
Fix a11y tab traversal

### PR DESCRIPTION
**Pull request by yjbanov**
Fix widget traversal when using the TAB key while a11y is enabled.

- Synchronize editing state that's reported through the semantics tree as opposed to that coming through the `flutter/textinput` channel.
- Add `tabindex=0` onto tappable targets, otherwise the browser doesn't treat the element as focusable.

This is an alternative approach to the previously proposed https://github.com/flutter/engine/pull/25401. This approach does not require changes on the framework side.

Fixes https://github.com/flutter/flutter/issues/52106

### CODE REVIEW

**Comment by yjbanov** [comment]
I'm going to debug the Mac Safari test but otherwise this is ready for review. I don't think the approach will change much.

File: lib/web_ui/lib/src/engine/semantics/text_field.dart [hunk 1]
```diff
@@ -47,28 +113,15 @@ class SemanticsTextEditingStrategy extends DefaultTextEditingStrategy {
     _subscriptions.clear();
     _lastEditingState = null;
 
-    // If focused element is a part of a form, it needs to stay on the DOM
```
Comment (ferhatb) [inline lib/web_ui/lib/src/engine/semantics/text_field.dart:50]: Looks like this was added as a fix for form autofill, does disabling keep form submit intact?

File: lib/web_ui/lib/src/engine/semantics/text_field.dart [hunk 2]
```diff
@@ -79,59 +132,63 @@ class SemanticsTextEditingStrategy extends DefaultTextEditingStrategy {
...
-    domElement.focus();
+    activeDomElement.focus();
+  }
+
+  @override
+  void initializeElementPlacement() {
+    // Element placement is done by [TextField].
   }
 
   @override
```
Comment (ferhatb) [inline lib/web_ui/lib/src/engine/semantics/text_field.dart:165]: remove empty override?

File: lib/web_ui/test/text_editing_test.dart [hunk 3]
```diff
@@ -335,270 +344,29 @@ void testMain() {
...
-      testInputElement = InputElement();
-      testTextEditing = HybridTextEditing();
-      editingElement = GloballyPositionedTextEditingStrategy(testTextEditing);
-    });
-
-    tearDown(() {
-      testInputElement = null;
-    });
-
-    test('autofill form lifecycle works', () async {
```
Comment (ferhatb) [inline lib/web_ui/test/text_editing_test.dart:385]: Is there an integration test for autofill that still covers this?

**Review by ferhatb — CHANGES_REQUESTED** [review]

File: lib/web_ui/lib/src/engine/dom_renderer.dart [hunk 4]
```diff
@@ -92,7 +92,7 @@ class DomRenderer {
   /// This getter calls the `hasFocus` method of the `Document` interface.
   /// See for more details:
   /// https://developer.mozilla.org/en-US/docs/Web/API/Document/hasFocus
-  bool? get windowHasFocus =>
+  bool get windowHasFocus =>
       js_util.callMethod(html.document, 'hasFocus', <dynamic>[]);
```
[the remark this answers is not in the document]
Comment (yjbanov) [reply lib/web_ui/lib/src/engine/dom_renderer.dart:96]: Done.

File: lib/web_ui/lib/src/engine/semantics/tappable.dart [hunk 5]
```diff
@@ -20,6 +20,10 @@ class Tappable extends RoleManager {
   void update() {
     final html.Element element = semanticsObject.element;
 
+    // "tab-index" HTML attribute is necessary to make the element can be
```
[the remark this answers is not in the document]
Comment (yjbanov) [reply lib/web_ui/lib/src/engine/semantics/tappable.dart:23]: Done.

File: lib/web_ui/lib/src/engine/semantics/tappable.dart [hunk 6]
```diff
@@ -48,6 +52,9 @@ class Tappable extends RoleManager {
         _stopListening();
       }
     }
+    if (semanticsObject.isFlagsDirty && semanticsObject.hasFocus) {
```
[the remark this answers is not in the document]
Comment (yjbanov) [reply lib/web_ui/lib/src/engine/semantics/tappable.dart:55]: Done.

File: lib/web_ui/lib/src/engine/semantics/text_field.dart (hunk 1 above)
Comment (yjbanov) [reply lib/web_ui/lib/src/engine/semantics/text_field.dart:50]: No, I removed form autofill from semantics. It was broken. The non-a11y version is still intact.

File: lib/web_ui/lib/src/engine/semantics/text_field.dart (hunk 2 above)
Comment (yjbanov) [reply lib/web_ui/lib/src/engine/semantics/text_field.dart:165]: The empty override makes these methods noop.

File: lib/web_ui/test/text_editing_test.dart (hunk 3 above)
Comment (yjbanov) [reply lib/web_ui/test/text_editing_test.dart:385]: I'm not changing tests for non-a11y form autofill. This is a11y autofill, which I'm removing due to broken design.

**Review by ferhatb — APPROVED** [review]

**Pull request merged**